packages/atclient/src/atclient.c

```diff
@@ -270,14 +270,136 @@ exit: { return ret; }
 }
 
 int atclient_get_publickey(const atclient *atclient, const atclient_atkey *atkey, char *value, const size_t valuelen,
-                           size_t *valueolen) {
+                           size_t *valueolen, bool bypasscache) {
   int ret = 1;
 
-  // TODO: implement
+  // 1. initialize variables
+  atclient_atstr cmdbuffer;
+  atclient_atstr_init_literal(&cmdbuffer, 4096, "plookup:");
+
+  atclient_atstr atkeystr;
+  atclient_atstr_init(&atkeystr, ATKEY_GENERAL_BUFFER_SIZE);
+
+  const size_t recvlen = 4096;
+  atclient_atbytes recv;
+  atclient_atbytes_init(&recv, recvlen);
+
+  cJSON *root = NULL;
+
+  if(atkey->atkeytype != ATCLIENT_ATKEY_TYPE_PUBLICKEY) {
```

Comment:
This could be checked just at the beginning, to avoid unnecessary computations (and do `return 1` instead of `goto exit`)

Reply:
https://github.com/atsign-foundation/at_c/pull/137/commits/bd109f75b7f13b6dcff00495b19002c7b23db678

Reply:
Resolving this,
thanks @realvarx